js/main.js

```diff
@@ -1,2 +1,34 @@
 /* global data */
 /* exported data */
+
+var $photoUrl = document.querySelector('#photolink');
+var $photo = document.querySelector('#photo');
+
+$photoUrl.addEventListener('input', handleInput);
+
+function handleInput() {
+  $photo.setAttribute('src', $photoUrl.value);
+  if ($photoUrl.value === '') {
+    $photoUrl.setAttribute('src', 'images/placeholder-image-square.jpg');
+  }
+}
+
+var $form = document.querySelector('#code-form');
+$form.addEventListener('submit', submitForm);
+
+function submitForm(event) {
+  event.preventDefault();
+  var newObj = {};
+  newObj.entryID = data.nextEntryId;
+  newObj.title = $form.title.value;
+  newObj.photourl = $form.photo.value;
+  newObj.notes = $form.notes.value;
+  data.nextEntryId++;
+  data.entries.push(newObj);
+  document.querySelector('#photo').setAttribute('src', 'images/placeholder-image-square.jpg');
+  $form.reset();
+  var dataJSON = JSON.stringify(data);
```

Comment:
make sure to move all local storage functionality into the `data.js` file, and replicate the functionality from the `js local storage` exercise, except save the entire data object in the local storage. Local storage should not be touched until the page refreshes.